tasks/kernel_matrix_testing/compiler.py

```diff
@@ -114,10 +113,44 @@ def start(self) -> None:
         self.exec("echo conda activate ddpy3 >> /home/compiler/.bashrc", user="compiler")
         self.exec(f"install -d -m 0777 -o {uid} -g {uid} /go", user="root")
 
-
-def get_compiler(ctx: Context, arch: ArchOrLocal):
-    return CompilerImage(ctx, full_arch(arch))
-
-
-def all_compilers(ctx: Context):
-    return [get_compiler(ctx, arch) for arch in arch_mapping.values()]
+        self.prepare_for_cross_compile()
+
+    def ensure_ready_for_cross_compile(self):
+        res = self.exec("test -f /tmp/cross-compile-ready", user="root", warn=True)
+        if res is not None and not res.ok:
+            info("[*] Compiler image not ready for cross-compilation, preparing...")
+            self.prepare_for_cross_compile()
+
+    def prepare_for_cross_compile(self):
+        target: Arch
+        if self.arch == ARCH_ARM64:
+            target = ARCH_AMD64
+        else:
+            target = ARCH_ARM64
+
+        # Hardcoded links to the header packages for each architecture. Why do this and not have something more automated?
+        # 1. While right now the URLs are similar and we'd only need a single link with variable replacement, this might
+        #    change as the repository layout is not under our control.
+        # 2. Automatic detection of these URLs is not direct (querying the package repo APIs is not trivial) and we'd need some
+        #    level of hard-coding some URLs or assumptions anyways.
+        # 3. Even if someone forgets to update these URLs, it's not a big deal, as we're building inside of a Docker image which will
+        #    likely have a different kernel than the target system where the built eBPF files are going to run anyways.
+        header_package_urls: dict[Arch, str] = {
+            ARCH_AMD64: "http://deb.debian.org/debian-security/pool/updates/main/l/linux-5.10/linux-headers-5.10.0-0.deb10.28-amd64_5.10.209-2~deb10u1_amd64.deb",
+            ARCH_ARM64: "http://deb.debian.org/debian-security/pool/updates/main/l/linux-5.10/linux-headers-5.10.0-0.deb10.28-arm64_5.10.209-2~deb10u1_arm64.deb",
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ As it is hardcoded, it would be better to have a constant at the top of this file IMO

Reply:
Changed!